core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts

```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
+        return languageId$.pipe(
+            switchMap((languageId) => this.languageService.makeDefault(languageId)),
+            switchMap(() => this.languageService.get()),
+            tap((languages) => {
+                this.setLocales(languages);
+                this.messageService.add({
+                    severity: 'success',
+                    summary: this.dotMessageService.get('locale.notification.success.title'),
+                    detail: this.dotMessageService.get('locale.notification.success.message')
+                });
+            }),
+            catchError((error: HttpErrorResponse) => this.dotHttpErrorManagerService.handle(error))
+        );
+    });
+
+    readonly deleteLocale = this.effect<number>((languageId$) =>
+        languageId$.pipe(
+            switchMap((languageId) => this.languageService.delete(languageId)),
+            switchMap(() => this.languageService.get()),
+            tap((languages) => {
+                this.setLocales(languages);
+                this.messageService.add({
+                    severity: 'info',
+                    summary: this.dotMessageService.get(
+                        'locale.delete.confirmation.notification.title'
+                    ),
+                    detail: this.dotMessageService.get(
+                        'locale.delete.confirmation.notification.message'
+                    )
+                });
+            }),
+            catchError((error: HttpErrorResponse) => this.dotHttpErrorManagerService.handle(error))
+        )
+    );
+
     constructor() {
-        super({ status: ComponentStatus.IDLE, locales: [] });
+        super({
+            status: ComponentStatus.IDLE,
+            locales: [],
+            initialLocales: [],
+            countries: [],
+            languages: [],
+            isEnterprise: false,
+            pushPublishEnvironments: []
+        });
     }
 
     /**
      * Private function to process the languages into the format needed for the state
      */
-    private processLanguages(languages: DotLanguage[]): DotLocaleRow[] {
-        return languages.map((language) => ({
-            locale: `${language.language} (${language.isoCode})`,
-            language: `${language.language} - ${language.languageCode}`,
-            country: `${language.country} - ${language.countryCode}`,
+    private processLanguages(
+        locales: DotLanguage[],
+        isEnterprise: boolean,
+        pushPublishEnvironments: DotEnvironment[]
+    ): DotLocaleRow[] {
+        return locales.map((locale) => ({
+            id: locale.id,
+            locale: `${locale.language} (${locale.isoCode})`,
+            language: `${locale.language} - ${locale.languageCode}`,
+            country: `${locale.country} - ${locale.countryCode}`,
             variables: 'TBD',
-            defaultLanguage: language.defaultLanguage,
+            defaultLanguage: locale.defaultLanguage,
             actions: [
```

Comment:
This seems like a lot of state here... should we this be his own state and component? We had this problem in Pages, try to add all the actions of the menus in the same store as the page and now is a big spagetti code.

Reply:
I don't think in this case; languages only have a max of 4 actions.